src/apps/dashboard/routes/settings/index.tsx

```diff
@@ -0,0 +1,266 @@
+        picker.show({
+            callback: function (path: string) {
+                if (path) {
+                    setCachePath(path);
+                }
+
+                picker.close();
+            },
+            validateWriteable: true,
+            header: globalize.translate('HeaderSelectServerCachePath'),
+            instruction: globalize.translate('HeaderSelectServerCachePathHelp')
+        });
+    }, []);
+
+    const showMetadataPathPicker = useCallback(() => {
+        const picker = new DirectoryBrowser();
+
+        picker.show({
+            path: metadataPath,
+            callback: function (path: string) {
+                if (path) {
+                    setMetadataPath(path);
+                }
+
+                picker.close();
+            },
+            validateWriteable: true,
+            header: globalize.translate('HeaderSelectMetadataPath'),
+            instruction: globalize.translate('HeaderSelectMetadataPathHelp')
+        });
+    }, [metadataPath]);
+
+    useEffect(() => {
+        if (!isSystemInfoPending && !isSystemInfoError) {
+            setCachePath(systemInfo.CachePath);
+            setMetadataPath(systemInfo.InternalMetadataPath);
+        }
+    }, [systemInfo, isSystemInfoPending, isSystemInfoError]);
+
+    if (isConfigPending || isLocalizationOptionsPending || isSystemInfoPending) {
+        return <Loading />;
+    }
+
+    return (
+        <Page
+            id='dashboardGeneralPage'
+            title={globalize.translate('General')}
+            className='type-interior mainAnimatedPage'
+        >
+            <Box className='content-primary'>
+                {isConfigError || isLocalizationOptionsError || isSystemInfoError ? (
+                    <Alert severity='error'>{globalize.translate('SettingsPageLoadError')}</Alert>
+                ) : (
+                    <Form method='POST'>
+                        <Stack spacing={3}>
+                            <Typography variant='h1'>{globalize.translate('Settings')}</Typography>
+
+                            {!isSubmitting && actionData?.isSaved && (
+                                <Alert severity='success'>
+                                    {globalize.translate('SettingsSaved')}
+                                </Alert>
+                            )}
+
+                            <TextField
+                                name='ServerName'
+                                label={globalize.translate('LabelServerName')}
+                                helperText={globalize.translate('LabelServerNameHelp')}
+                                defaultValue={systemInfo.ServerName}
+                            />
+
+                            <TextField
+                                select
+                                name='UICulture'
+                                label={globalize.translate('LabelPreferredDisplayLanguage')}
+                                helperText={(
+                                    <Stack>
```

Comment:
Using `Stack` here is causing an error to be logged:
```
Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.
```

Maybe just use a `<br />` between the text and link?